djes/search.py

```diff
@@ -52,11 +52,27 @@ def hits(self):
 class LazySearch(Search):
     """This extends the base Search object, allowing for Django-like lazy execution"""
 
+    def __init__(self, *args, **kwargs):
+        super(LazySearch, self).__init__(*args, **kwargs)
+        self.position = kwargs.get("from", 0)
+
     def __len__(self):
         if 'size' in self._extra:
             return min(self._extra['size'], self.count())
         return self.count()
 
+    def __next__(self):
+        return self.next()
+
+    def next(self):
+        # Tracks index on the search object so we can iterate across multiple queries.
+        start, end = self.position, self.position + 1
+        self.position += 1
+        try:
+            return self[start:end][0]
```

Comment:
Would the second next() call trigger another ES request? Or would it use the cached search[:10] results